tensorly/regression/cp_plsr.py

```diff
@@ -0,0 +1,259 @@
+                if Z.ndim >= 2:
+                    Z_comp = tucker(Z, [1] * T.ndim(Z))[1]
+                else:
+                    Z_comp = [Z / T.norm(Z)]
+                for mode in range(1, X.ndim):  # First mode of Z is collapsed by the above tensordot call
+                    comp_X_factors[mode] = T.reshape(Z_comp[mode - 1], (-1,))
+
+                comp_X_factors[0] = multi_mode_dot(
+                    X, comp_X_factors[1:], range(1, T.ndim(X))
+                )
+                comp_Y_factors_1 = T.dot(T.transpose(Y), comp_X_factors[0])
+                comp_Y_factors_1 /= T.norm(comp_Y_factors_1)
+                comp_Y_factors_0 = T.dot(Y, comp_Y_factors_1)
+
+                if T.norm(old_comp_Y_factors_0 - comp_Y_factors_0) < self.tol:
+                    if self.verbose:
+                        print(f"Component {component}: converged after {iter} iterations")
+                    break
+                old_comp_Y_factors_0 = T.copy(comp_Y_factors_0)
+
+            # Put iteration results back to the parameter variables
+            for ii in range(len(comp_X_factors)):
+                self.X_factors[ii] = T.index_update(
+                    self.X_factors[ii], T.index[:, component], comp_X_factors[ii]
+                )
+            self.Y_factors[0] = T.index_update(
+                self.Y_factors[0], T.index[:, component], comp_Y_factors_0
+            )
+            self.Y_factors[1] = T.index_update(
+                self.Y_factors[1], T.index[:, component], comp_Y_factors_1
+            )
+
+            # Deflation
+            X -= CPTensor(
+                (None, [T.reshape(ff, (-1, 1)) for ff in comp_X_factors])
+            ).to_tensor()
+            Y -= T.dot(
+                T.dot(
+                    self.X_factors[0],
+                    T.lstsq(self.X_factors[0], T.reshape(comp_Y_factors_0, (-1, 1)))[0],
+                ),
+                T.reshape(comp_Y_factors_1, (1, -1)),
+            )  # Y -= T pinv(T) u q'
+
+        return self
+
+    def predict(self, X):
+        """Returns the predicted labels for a new data tensor
+
+        Parameters
+        ----------
+        X : ndarray
+            tensor data of shape (n_samples, N1, ..., NS)
+        """
+        if self.X_shape_[1:] != T.shape(X)[1:]:
+            raise ValueError(
+                f"Training X has shape {self.X_shape_}, while the new X has shape {T.shape(X)}"
+            )
+        X -= self.X_mean_
+        factors_kr = khatri_rao(self.X_factors, skip_matrix=0)
+        unfolded = unfold(X, 0)
+        scores = T.lstsq(factors_kr, T.transpose(unfolded))[0]  # = Tnew
+        estimators = T.lstsq(self.X_factors[0], self.Y_factors[0])[0]
+        return (
+            T.dot(
+                T.dot(T.transpose(scores), estimators), T.transpose(self.Y_factors[1])
+            )
+            + self.Y_mean_
+        )
+
+    def transform(self, X, Y=None):
+        """Apply the dimension reduction.
```

Comment:
Would be useful to have a slightly longer, more informative description for users that are going to use this as a black-box :)